llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp

```diff
@@ -454,6 +454,10 @@ amdhsa::kernel_descriptor_t AMDGPUAsmPrinter::getAmdhsaKernelDescriptor(
 }
 
 bool AMDGPUAsmPrinter::runOnMachineFunction(MachineFunction &MF) {
+  if (MF.getFunction().hasFnAttribute("amdgpu-lib-fun") &&
+      !MF.getFunction().hasFnAttribute("amdgpu-backend-used"))
```

Comment:
Can't `llvm.used` be used instead of a new ad-hoc `amdgpu-backend-used` attribute?
https://llvm.org/docs/LangRef.html#the-llvm-used-global-variable

Also where is `amdgpu-lib-fun` added? If users are expected to add it, it should be documented in `AMDGPUUsage.rst`